### DISCUSSION TITLE
Is there a way to horizontally and vertically center a toast?

### DISCUSSION THREAD

**Discussion by daweilv** [chakra-ui/chakra-ui#7388] [Q&A]
I've read through the doc, can not find a way to horizontally and vertically center a toast. Can anyone help me? 

`const positions = [
    'top',
    'top-right',
    'top-left',
    'bottom',
    'bottom-right',
    'bottom-left',
  ]`

**Comment by keithburgie** [comment]
Do you mean that you're trying to put it in the center of the screen? Could you describe where you'd like one of these examples to go?

![Screenshot 2023-02-23 at 8 53 42 AM](https://user-images.githubusercontent.com/96078669/220927741-587f98ba-aafd-42c5-bf49-5fc5473c2e46.png)

**Reply by daweilv** [reply]
<img width="383" alt="image" src="https://user-images.githubusercontent.com/7405300/221418248-11dcd5ae-ea9f-49ef-9cc9-7816d9d5cf89.png">
Hi, 
Case 1: In e-commerce scenarios, we often send out coupons, and displaying them in the center on mobile devices makes them more prominent.

Case 2: When loading content on the internet, we often add a loading state, which is also centered on the screen.

**Reply by keithburgie** [reply]
Could you accomplish your goal with one of these other components?

Alert
https://chakra-ui.com/docs/components/alert

Alert Dialog
https://chakra-ui.com/docs/components/alert-dialog

Modal
https://chakra-ui.com/docs/components/modal

I hear you on the loading spinner. For something like that, I would use something like an absolutely positioned div with a spinner inside it. You could put a spinner inside a modal, for example, and then set the modal background, shadow, etc. to transparent.

I think that toasts are absolutely or fixed positioned and set dynamically with JavaScript. It might be an uphill battle to try to center one. You could try that strategy of putting one inside a container that you have better control over, though. It won't work if toasts are fixed positioned.
